src/graphql/schema/collection/createCollectionGqlType.ts

```diff
@@ -45,12 +46,17 @@ export default function createCollectionGqlType<TValue> (
       Array.from(type.fields).map(
         <TFieldValue>([fieldName, field]: [string, ObjectType.Field<TValue, TFieldValue>]) => {
           const { gqlType, intoGqlOutput } = getGqlOutputType(buildToken, field.type)
+          const formattedFieldName = formatName.field(fieldName)
           return {
-            key: formatName.field(fieldName),
+            key: formattedFieldName
             value: {
               description: field.description,
               type: gqlType,
-              resolve: (value: TValue): mixed => intoGqlOutput(field.getValue(value)),
+              sqlName: () => fieldName,
```

Comment:
> but I think a full two-stage parse of the entire AST is a bit ambitious at this stage

The more I think about it, you’re probably right for this case. What gets really tricky is if we have procedure fields. Do we call that procedure for every object of that type? We really shouldn’t…

Instead, we should see if there is anything we can do to create effectively one data loader for every field. One data loader per request turns out to be hard for us to use here, but one data loader per field could work well.